Updated application and application preface

RandyHuynh5815 opened the pull request:
Made changes to the application and application preface to clarify the application process and acknowledgments
- Removed "Optional" from Social Information header and put them on the portfolio and LinkedIn link inputs (clarify that resume is required)
- Added required validation for resume field
- Swapped the two prompts so the required prompt is first
- Added mentor/volunteer links on the preface
- Change login button to say "Log In To Apply"
- Added application deadline in bold

TODOs:
- Get input from Sydnee (or anyone from logistics) about wording changes on the preface

Resolves #62

RandyHuynh5815 committed:
Removed misleading aspects of application

RandyHuynh5815 committed:
Update application preface

samderanova reviewed (commented):
Thanks for picking this up! Were you able to test it to see if it works properly? Or do we have to push to staging in order to see?

RandyHuynh5815 commented:
I tested the front-end portion of the application (the validation is showing up and submission being prevented rather than throwing a form submission failure without a resume added). However, I do not have the authorization to fully test the response going into the backend as well. That means we would have to push to staging if we want an end-to-end test or test on your end if you have the authorization to do so.

samderanova commented:
I actually don't have `SP_KEY` so I can't test anything on my end....

taesungh commented:
We went with making resume optional with #51, so we can revert the required check.

RandyHuynh5815 committed:
Revert changes so resume is optional

taesungh committed:
Fix grammar in application preface

taesungh committed:
Add rewrites for mentor and volunteer application

- Redirect `/mentor` and `/volunteer` to corresponding Google Forms
- Replace referenced links

rileysw commented on src/views/apply/components/ApplySections/SocialInformation.tsx:
```diff
@@ -19,8 +19,20 @@ function SocialInformation({
 	return (
 		<div>
-			<h3>Social Information (Optional)</h3>
+			<h3>Social Information (Optional Section)</h3>
```
Let's take out `(Optional Section)` since the bolded text below already tells us that.

taesungh replied:
Fixed in 76b73fe.

samderanova commented on src/views/home/sections/MentorsVolunteers/MentorsVolunteers.tsx:
```diff
@@ -29,7 +29,8 @@ function MentorsVolunteers() {
 								variant="museum"
-								href="https://forms.gle/tcnikpj5gHnGPNvx7"
+								href="/mentor"
+								role=""
```
Clarification: The default value for `role` is `button` right? Is that why it's set to an empty string here?

taesungh replied:
Yeah, in this case, the actual wording of the button text would be more appropriate for a link (default role for `<a>`). This would not be necessary for the "Apply as ..." buttons in the Home landing since those are worded to be "actionable".

rileysw commented on src/views/home/sections/Landing.tsx:
```diff
@@ -14,10 +14,7 @@ function Landing() {
-					className="btn-museum btn-landing"
-					href="https://forms.gle/tcnikpj5gHnGPNvx7"
-				>
+				<Button className="btn-museum btn-landing" href="/mentor">
```
These rewrites aren't working on my end. Is it supposed to normally redirect to the google form page?

taesungh replied:
Did you restart the development server when pulling these changes? If you didn't, it should tell you
> Found a change in next.config.js. Restart the server to see the changes in effect.

rileysw replied:
Oop, looks like I didn't, thanks Taesung.

taesungh committed:
Remove redundant Optional in SocialInformation

- Change optional description to muted
